### PR TITLE
Tasks: Add video-text-to-text task page 

### PR DESCRIPTION
**Pull request by merveenoyan**
cc @zucchini-nlp @NielsRogge

### CODE REVIEW

File: packages/tasks/src/tasks/video-text-to-text/data.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		},
+        {
+			description: "Large and powerful video-text-to-text model that can take in image and video inputs.",
+			id: "llava-hf/LLaVA-NeXT-Video-34B-hf",
+		},
+	],
+	spaces: [
+		{
+			description: "An application to chat with a video-text-to-text model.",
+			id: "llava-hf/video-llava",
```
Comment (zucchini-nlp) [inline packages/tasks/src/tasks/video-text-to-text/data.ts:50]: not sure if this should be a real model-id, but video-llava is under LanguageBind org in the hub

File: packages/tasks/src/tasks/video-text-to-text/data.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+					"The video shows a series of images showing a fountain with water jets and a variety of colorful flowers and butterflies in the background.",
+				type: "text",
+			},
+		],
+	},
+	metrics: [],
+	models: [
+		{
+			description: "A robust video-text-to-text model that can take in image and video inputs.",
+			id: "llava-hf/llava-interleave-qwen-7b-hf",
```
Comment (zucchini-nlp) [inline packages/tasks/src/tasks/video-text-to-text/data.ts:40]: can we change this to llava-onevision, that is better than interleave and I'll merge the PR for it soon

File: packages/tasks/src/tasks/video-text-to-text/data.ts (hunk 1 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/video-text-to-text/data.ts:50]: this isn't a model id, it is a space

File: packages/tasks/src/tasks/video-text-to-text/about.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
+Most of the video language models can take in videos, multiple videos, images and multiple images. Some of these models can also take interleaved inputs, which can have images and videos inside the text, where you can refer to the input image and input video within the text prompt.
+
+## Different Types of Vide Language Models
```
Comment (julien-c) [inline packages/tasks/src/tasks/video-text-to-text/about.md:3]:
```suggestion
## Different Types of Video Language Models
```

File: packages/tasks/src/tasks/video-text-to-text/about.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+model = LlavaNextVideoForConditionalGeneration.from_pretrained(
+    model_id, 
+    torch_dtype=torch.float16, 
+    low_cpu_mem_usage=True, 
+).to(device)
+
+processor = LlavaNextVideoProcessor.from_pretrained(model_id)
+
+def sample_frames(url, num_frames):
+
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/about.md:50]:
```suggestion
```

File: packages/tasks/src/tasks/video-text-to-text/about.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    response = requests.get(url)
+    path_id = str(uuid.uuid4())
+
+    path = f"./{path_id}.mp4"
+
+    with open(path, "wb") as f:
+      f.write(response.content)
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/about.md:57]:
```suggestion
         f.write(response.content)
```

The rest of indents in the snippet use 4 spaces

File: packages/tasks/src/tasks/video-text-to-text/about.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+## Inference
+
+You can use the Transformers library to interact with video-language models. 
+Below we load [a video language model](https://huggingface.co/llava-hf/LLaVA-NeXT-Video-7B-hf), write a simple utility to sample videos, use chat template to format the text prompt, process the video and the text prompt and infer. 
+
+```python
+import uuid
+import requests
+import cv2
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/about.md:34]: maybe mention that you need to install opencv-python? The other dependencies are fine imo.

File: packages/tasks/src/tasks/video-text-to-text/about.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    total_frames = int(video.get(cv2.CAP_PROP_FRAME_COUNT))
+    interval = total_frames // num_frames
+    frames = []
+    for i in range(total_frames):
+        ret, frame = video.read()
+        pil_img = Image.fromarray(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))
+        if not ret:
+            continue
+        if i % interval == 0:
+            frames.append(pil_img)
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/about.md:69]:
```suggestion
        if not ret:
            continue
        if i % interval == 0:
             pil_img = Image.fromarray(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))
            frames.append(pil_img)
```

File: packages/tasks/src/tasks/video-text-to-text/data.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		},
+	],
+	spaces: [
+		{
+			description: "An application to chat with a video-text-to-text model.",
+			id: "llava-hf/video-llava",
+		},
+	],
+	summary:
+		"Video-text-to-text models take in an video and text prompt and output text. These models are also called video-language models.",
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/data.ts:54]:
```suggestion
		"Video-text-to-text models take in a video and a text prompt and output text. These models are also called video-language models.",
```

File: packages/tasks/src/tasks/video-text-to-text/data.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+			description: "A dataset of instructions and question-answer pairs about videos.",
+			id: "lmms-lab/VideoChatGPT",
+		},
+		
+	],
+	demo: {
+		inputs: [
+			{
+				filename: "video-text-to-text-input.gif",
+				type: "video",
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/data.ts:19]: This type does not exist [here](https://github.com/huggingface/huggingface.js/blob/4959181d3c43ad5deb1c2a3a05db94abfe358dbe/packages/tasks/src/tasks/index.ts#L254). I don't know if it should be added in https://github.com/huggingface-internal/moon-landing/blob/main/server/views/components/TaskSchemaEntry/TaskSchemaEntry.svelte and, in that case, if it needs a new widget type.

As an alternative, we could use `"img"`, but not sure if the `gif` will be rendered. Or prepare a static image that shows a few frames.

What do you think @osanseviero?

Comment (merveenoyan) [reply packages/tasks/src/tasks/video-text-to-text/data.ts:19]: ah yes sorry it should be image

File: packages/tasks/src/tasks/video-text-to-text/data.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+		{
+			description: "Multiple-choice questions and answers about videos.",
+			id: "lmms-lab/Video-MME",
+		},
+        {
+			description: "A dataset of instructions and question-answer pairs about videos.",
+			id: "lmms-lab/VideoChatGPT",
+		},
+		
+	],
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/data.ts:14]:
```suggestion
		},
		{
			description: "A dataset of instructions and question-answer pairs about videos.",
			id: "lmms-lab/VideoChatGPT",
		},	
	],
```

File: packages/tasks/src/tasks/video-text-to-text/data.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+			},
+		],
+	},
+	metrics: [],
+	models: [
+		{
+			description: "A robust video-text-to-text model that can take in image and video inputs.",
+			id: "llava-hf/llava-onevision-qwen2-72b-ov-hf",
+		},
+        {
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/data.ts:42]:
```suggestion
		},
		{
```

**Review by pcuenca — COMMENTED** [review]
Linter suggestions

**Comment by merveenoyan** [comment]
IDK why linting on my local results in no change but here it fails

**Comment by merveenoyan** [comment]
I did another passthrough and couldn't see anything that I'd change, would love an approval if there's no other thing y'all would change here

File: packages/tasks/src/tasks/video-text-to-text/about.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
+Most of the video language models can take in videos, multiple videos, images and multiple images. Some of these models can also take interleaved inputs, which can have images and videos inside the text, where you can refer to the input image and input video within the text prompt.
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/about.md:1]:
```suggestion
Most of the video language models can take in videos, multiple videos, images and multiple images. Some of these models can also take interleaved inputs, which can have images and videos inside the text, where you can refer to the input images and input videos within the text prompt.
```

File: packages/tasks/src/tasks/video-text-to-text/about.md [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+
+Video language models trained on video-question-answer pairs can be used for video question answering and generating captions for videos.
+
+### Video Chat
+
+Video language models can be used to have a dialogue about a video.
+
+### Video Recognition with Instructions
+
+Video language models can recognize images through descriptions. When given detailed descriptions of specific entities, it can classify the entities in a video.
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/about.md:23]:
```suggestion
Video language models can recognize images through descriptions. When given detailed descriptions of specific entities, they can classify the entities in a video.
```

File: packages/tasks/src/tasks/video-text-to-text/about.md [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+
+## Inference
+
+You can use the Transformers library to interact with video-language models.
+Below we load [a video language model](https://huggingface.co/llava-hf/LLaVA-NeXT-Video-7B-hf), write a simple utility to sample videos, use chat template to format the text prompt, process the video and the text prompt and infer. To run below snippet, please install [OpenCV](https://pypi.org/project/opencv-python/) by running `pip install opencv-python`.
```
Comment (pcuenca) [inline packages/tasks/src/tasks/video-text-to-text/about.md:28]:
```suggestion
Below we load [a video language model](https://huggingface.co/llava-hf/LLaVA-NeXT-Video-7B-hf), write a simple utility to sample videos, use chat template to format the text prompt, process the video and the text prompt and infer. To run the snippet below, please install [OpenCV](https://pypi.org/project/opencv-python/) by running `pip install opencv-python`.
```

**Review by pcuenca — APPROVED** [review]
Did a final pass and suggested a couple of minor nits, LGTM 🔥

**Pull request merged**